src/components/common/Members/Members.tsx

```diff
@@ -0,0 +1,204 @@
+
+import { useColonyContext } from '~hooks';
+import { Domain, useGetMembersForColonyQuery, SortingMethod } from '~gql';
+import {
+  COLONY_TOTAL_BALANCE_DOMAIN_ID,
+  ALLDOMAINS_DOMAIN_SELECTION,
+  ROOT_DOMAIN_ID,
+} from '~constants';
+import { User } from '~types';
+import { notNull } from '~utils/arrays';
+
+import MembersTitle from './MembersTitle';
+// import { filterMembers } from './filterMembers';
+import MembersSection from './MembersSection';
+
+import styles from './Members.css';
+
+const displayName = 'common.Members';
+
+const MSG = defineMessages({
+  loading: {
+    id: `${displayName}.loading`,
+    defaultMessage: "Loading Colony's users...",
+  },
+  failedToFetch: {
+    id: `${displayName}.failedToFetch`,
+    defaultMessage: "Could not fetch the colony's members",
+  },
+  noMembersFound: {
+    id: `${displayName}.noMembersFound`,
+    defaultMessage: 'No members found',
+  },
+});
+
+interface Props {
+  selectedDomain: number | undefined;
+  handleDomainChange: React.Dispatch<React.SetStateAction<number>>;
+  filters: any;
+  // filters: FiltersFormValues;
+}
+
+export type Member = User & {
+  roles: ColonyRole[];
+  directRoles: ColonyRole[];
+  banned: boolean;
+};
+
+const Members = ({ selectedDomain, handleDomainChange, filters }: Props) => {
+  const { colony } = useColonyContext();
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [sortingMethod, setSortingMethod] = useState(
+    SortingMethod.ByHighestRep,
+  );
+
+  const { data, loading: loadingMembers } = useGetMembersForColonyQuery({
+    skip: !colony?.colonyAddress,
+    variables: {
+      input: {
+        colonyAddress: colony?.colonyAddress ?? '',
+        domainId: selectedDomain || COLONY_TOTAL_BALANCE_DOMAIN_ID,
+        sortingMethod,
+      },
+    },
+    fetchPolicy: 'cache-and-network',
+  });
+
+  const contributors = useMemo(
+    () => data?.getMembersForColony?.contributors?.filter(notNull) ?? [],
+    [data],
+  );
+  const watchers = useMemo(
+    () => data?.getMembersForColony?.watchers?.filter(notNull) ?? [],
+    [data],
+  );
+
+  // const canAdministerComments =
+  //   hasRegisteredProfile &&
+  //   (hasRoot(currentUserRoles) || canAdminister(currentUserRoles));
+
+  const domainSelectOptions = sortBy(
+    [...(colony?.domains?.items || []), ALLDOMAINS_DOMAIN_SELECTION].map(
+      ({ nativeId, name }: Domain) => {
+        return {
+          value: nativeId?.toString(),
+          label: name || '',
+        };
+      },
+    ),
+    ['value'],
+  );
+
+  // handles search values & close button
+  const handleSearch = useCallback(
+    (event) => {
+      const value = event.target?.value || '';
+      setSearchValue(value);
+    },
+    [setSearchValue],
+  );
+
+  const isRootDomain = useMemo(
+    () =>
+      selectedDomain === ROOT_DOMAIN_ID ||
+      selectedDomain === COLONY_TOTAL_BALANCE_DOMAIN_ID,
+    [selectedDomain],
+  );
+
+  const membersContent = useMemo(() => {
```

Comment:
I feel that this should really be a stand-alone component, e.g. `MembersContent`, in its own file